### PR TITLE
Add hostport to use qrcp in local server with reverse proxy

### LINKED ISSUE #169: Hide port in QRCode to use qrcp behind a reverse proxy
I'm opening this issue because:

- [ ] I have found a bug
- [x] I want to request a feature
- [ ] I have a question
- [ ] Other

Feature proposal:

Add an option allowing users to hide the port (and the colon) displayed in the QRCode and URL, allowing people to put qrcp behind a reverse proxy.

Problem example:

1. User creates a reverse proxy entry for qrcp.example.com and forward it to localhost:6666;
2. User configs qrcp to use port 6666 and fqdn qrcp.example.com;
3. The generated QRCode redirects to http://qrcp.example.com:6666 instead of qrcp.example.com (:80).

Proposal:

- Add an option to hide the port and colon in the displayed QRCode, or
- Allow users to specify a full base URL. This option must conflict with `--fqdn` and `--port` option, and could even let users specify the protocol (example: `-f https://qrcp.example.com`.

Advantages:

- Cleaner URL;
- Allows HTTPS if the reverse proxy handle certificate generation (like Caddy);
- Allows people behind high-restricted proxy (not allowing users to access websites that are not on port 80 or 443), to use qrcp;
- Using a custom DNS + reverse proxy like presented, allows the use of qrcp behind NATed network (qrcd server is in a NATed network, reverse proxy is on the edge of the network or on a remote server forwarding traffic through SSH tunnel, and user is on a different network like 4G).

### PR DESCRIPTION
**Pull request by mattn**
Added hostport option to display really acceptable host:port via reverse proxy.

Fixes #169

If you think that FQDN is that and hostport should be passed to listen instead, please let me known.

### CODE REVIEW

**Comment by claudiodangelis** [comment]
Hello @mattn, thanks for this contribution. I didn't have time to review it yet, will hopefully do over the weekend.

Thanks,
C

**Comment by claudiodangelis** [comment]
I'm not too much convinced about the name of the flag.
Considering the following cases, I would rather go with something like `--baseurl`:

| Case | Args | Printed URL |
| --- | --- | --- |
| <ul><li>no reverse proxy</li></ul> | `-i tun0 -p 8080` | http://10.8.0.1:8080/send/A1s3 |
| <ul><li>no reverse proxy</li></ul> | `-i tun0 -p 8080` | http://10.8.0.1:8080/send/A1s3 |
| <ul><li>no reverse proxy</li><li>tls certs</li></ul> | `--tls-(cert,key...) -i tun0 -p 8080` | https://10.8.0.1:8080/send/A1s3 |
| <ul><li>reverse proxy</li><li>target is localhost:8080</li></ul>  | `--baseurl http://example.com -i lo -p 8080` | http://example.com/send/A1s3 |
| <ul><li>reverse proxy</li><li>target is localhost:8080</li><li>target on custom location</li></ul>  | `--baseurl http://example.com/qrcp -i lo -p 8080` | http://example.com/qrcp/send/A1s3 |
| <ul><li>reverse proxy</li><li>reverse proxy port 3000</li><li>target is localhost:8080</li></ul>  | `--baseurl http://example.com:3000 -i lo -p 8080` | http://example.com:3000/send/A1s3 |

Basically, if `baseurl` is passed, other parameters that control the printed URL (i.e.: `--secure`) should be ignored.
What do you think?

**Comment by claudiodangelis** [comment]
I will close this for now.
